### ISSUE TITLE
Replace "Home" with "Compose"

### ISSUE THREAD

**Issue by skx** [skx/webmail#7]
This would allow composing a fresh mail.

**Comment by MFAshby** [comment]
Regarding replying, I think the simplest thing to actually work in the frontend could just be  an HTML form element that sends plain-text only responses.

Nice things to include, I think: 
1. Quote the original message in the response (this is quite typical behaviour in most email clients) 
2. Formatted messages. Practically this might mean including a rich text editing component such as editorjs.io (there are many, many alternatives) 
3. Saving in drafts (automatically?) 

The backend presents a bit more difficulty. We would need the user to enter the SMTP(s) address and credentials on the login page as well, since these could be different from the IMAP login.

**Comment by MFAshby** [comment]
I've got a branch [here](https://github.com/MFAshby/webmail/compare/dedupe-html...MFAshby:compose?expand=1) where I'm working on adding the compose functionality. So far this just modifies the login page to add SMTP connection details. It's a bit crude, it just expects 3 more fields on the login page. These are defaulted to the same same as the SMTP credentials, but I expect this won't work in a lot of cases where host names for IMAP and SMTP are different.

It might be nice to make these _optional_ and only enable the 'compose' function when SMTP credentials are added. Maybe even prompt for SMTP credentials on demand when I actually compose an email, rather than upfront?

Next up will be a simple plain-text compose and send.

**Comment by skx** [comment]
Yes I tied myself in knots over this.  The best idea I had was to store them in a session, to avoid forcing them to be added to the login-screen.  On-compose you can prompt for them if missing.

Otherwise I figured we could simplify the login to allow a type of server to be chosen:

* username:
* password
* Provider: gmail, yahoo, aol,...  - Dropdown box 

With a "custom" option to allow you to enter imap/imaps/smtps/smtp host explicitly.  I prefer this idea to keep the user-experience nice and simple, but it doesn't help so much if the user is using a provider we don't list/include.

**Comment by MFAshby** [comment]
That makes sense, I think thunderbird does it that way. In fact I think thunderbird have an open database for settings for different providers which we might use https://support.mozilla.org/en-US/kb/automatic-account-configuration

I think some providers also use DNS SRV records, but I don't know how common this is in practice. https://tools.ietf.org/id/draft-daboo-srv-email-05.html#rfc.toc
